youtube_dl/extractor/archiveorg.py

```diff
@@ -52,7 +55,7 @@ def get_optional(metadata, field):
         metadata = self._download_json(
             'http://archive.org/details/' + video_id, video_id, query={
                 'output': 'json',
-            })['metadata']
+            }).get('metadata', {})
```

Comment:
Point of this?

Reply:
The idea is to make the extraction more resilient against website changes. With metadata being optional for downloading videos, it's probably better not to crash if the metadata element changes its name.

Reply:
This is not enough. If you reconsider this optional then you must also:
1. Handle non dict metadata scenario.
2. Handle download failure.

Reply:
You are right. Thanks. Commit 1326a5a fixes this by setting `fatal=False` when calling `_download_json()` and by checking the return value.